content/How to Make My First Gem - Timothy Peraza.md

```diff
@@ -0,0 +1,128 @@
+
+```ruby
+require 'blog_gem/version'
+
+module BlogGem
+  class Error < StandardError; end
+  # Your code goes here...
+end
+
+```
+
+As we can see, it is a class where, in principle, you could put code directly. That decision is up to the developer's wish, but generally no code is added to this file, instead of it, 'requires' are done to the files which contain the classes which offer the methods of our gem. We can also see this class does generic error handling which follows the ruby guidelines.
+
+## Spec folder
+
+<img src="images/spec_folder.png"
+     alt="Spec folder"
+     width="500" height="250"
+     align="middle"/>
+
+Both in Rootstrap and in the software world it is considered that good software always has to have unit tests which support the implemented code, for this very reason we use rspec. In the spec folder, we will put all the test files, where the unit tests of our gem will be implemented and which we can execute from the console with the rspec command.
+
+## Gemspec file
+
+```ruby
+require_relative 'lib/blog_gem/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = 'blog_gem'
+  spec.version       = BlogGem::VERSION
+  spec.authors       = ['TimoPeraza']
+  spec.email         = ['timothy@rootstrap.com']
+
+  spec.summary       = ''
+  spec.description   = ''
+  spec.homepage      = ''
+  spec.license       = 'MIT'
+  spec.required_ruby_version = Gem::Requirement.new('>= 2.3.0')
+
+  # spec.metadata["allowed_push_host"] = "TODO: Set to 'http://mygemserver.com'"
+
+  # spec.metadata["homepage_uri"] = spec.homepage
+  # spec.metadata["source_code_uri"] = "TODO: Put your gem's public repo URL here."
+  # spec.metadata["changelog_uri"] = "TODO: Put your gem's CHANGELOG.md URL here."
+
+  # Specify which files should be added to the gem when it is released.
+  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
+  spec.files = Dir['LICENSE.txt', 'README.md', 'lib/**/*']
+  spec.bindir        = 'exe'
+  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+  spec.require_paths = ['lib']
+  spec.add_development_dependency 'rake'
+  spec.add_development_dependency 'reek'
+  spec.add_development_dependency 'rspec'
+  spec.add_development_dependency 'rubocop'
+  spec.add_development_dependency 'simplecov', '~> 0.17.1'
+end
+```
+
+In this file we will put all the information about our gem, who made it, contact email, description, version, the files that compose the gem to do a release and also add all the external dependencies which the gem needs.
+
+By default, rsgem adds the dev dependencies: rake, reek, rspec, rubocop, and simplecov. This is because in Rootstrap we focus a lot on the quality of the code, it seems to us something very important when developing therefore, we add these dependencies to control that our gem has good quality.
```

Comment:
```suggestion
By default, Rsgem adds the following dev dependencies: Rake, Reek, RSpec, Rubocop, and Simplecov. We do this because at Rootrsap we like to focus on the qualify of the code, and as a result, we add these dependencies to ensure our gem has the highest quality possible.
```